common/lib/common-utils/src/test/mocha/typedEventEmitter.spec.ts

```diff
@@ -48,4 +48,72 @@ describe("TypedEventEmitter", () => {
         assert.equal(newListenerCalls, 1);
         assert.equal(removeListenerCalls, 1);
     });
+
+    it("Validate emit typing for valid event type", () => {
+        const eventArgsEmitted: any[] = [];
+        const handler = (...args): void => {
+            eventArgsEmitted.push(args);
+        };
+        const tee = new (class SomeClass extends TypedEventEmitter<ISampleEvents> {
+            someMember: 0 = 0;
+        })();
+        const plainTee = new TypedEventEmitter();
+
+        tee.on("noArgs", handler);
+        tee.on("twoArgs", handler);
+        tee.on("useThis", handler);
+        tee.on("somethingElse", handler);
+
+        tee.emit("noArgs");
+        tee.emit("twoArgs", true, "hello");
+        tee.emit("useThis", tee);
+
+        // @ts-expect-error Unknown event
+        tee.emit("somethingElse");
+        // @ts-expect-error Unknown event
+        tee.emit("noArgs", "bogus");
+        // @ts-expect-error Wrong arg types
+        tee.emit("twoArgs", "wrongType", 123);
```

Comment:
One more potential test case: `("twoArgs", true)` (correct type of arguments but not enough of them).